File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int n=s.length();
        unordered_map<char,int> mp;
        int left=0;
        int right=0;
        int maxx=0;
        while(right<n){
            mp[s[right]]++;
            while(mp[s[right]]>2){
                mp[s[left]]--;
                if(mp[s[left]]==0){
                    mp.erase(s[left]);
                }
                left++;
            }
            maxx=max(maxx,right-left+1);
            right++;
        }
        return maxx;
        
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
from collections import Counter
class Solution(object):
    def maximumLengthSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        mp=Counter()
        l=0
        r=0
        n=len(s)
        maxx=0
        while(r<n):
            mp[s[r]]+=1
            while(mp[s[r]]>2):
                mp[s[l]]-=1
                if(mp[s[l]]==0):
                   del mp[s[l]]
                l+=1
            maxx=max(maxx,r-l+1)
            r+=1
        return maxx

        